lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -419,10 +419,23 @@ public TokenDetails authorize(TokenParams params, AuthOptions options) throws Ab
 		options = (options == null) ? this.authOptions : options.copy();
 		params = (params == null) ? this.tokenParams : params.copy();
 
-		TokenDetails tokenDetails = tokenAuth.authorize(params, options, /*force=*/true);
+		try {
+			TokenDetails tokenDetails = tokenAuth.authorize(params, options, /*force=*/true);
+			ably.onAuthUpdated();
 
-		ably.onAuthUpdated();
-		return tokenDetails;
+			return tokenDetails;
+		}
+		catch (AblyException e) {
+			if(e.errorInfo != null) {
```

Comment:
What happens if `e.errorInfo == null` ? Can that happen? We need to call `onAuthError()` in any case.